js/src/client.ts

```diff
@@ -1597,34 +1597,55 @@ export class Client implements LangSmithTracingClientInterface {
     context: string,
     options?: { apiKey?: string; apiUrl?: string }
   ) {
-    try {
-      // Create multipart form data boundary
-      const boundary =
-        "----LangSmithFormBoundary" + Math.random().toString(36).slice(2);
-      const body = await (_globalFetchImplementationIsNodeFetch()
-        ? this._createNodeFetchBody(parts, boundary)
-        : this._createMultipartStream(parts, boundary));
-
-      const headers: Record<string, string> = {
-        ...this.headers,
-        "Content-Type": `multipart/form-data; boundary=${boundary}`,
-      };
-      if (options?.apiKey !== undefined) {
-        headers["x-api-key"] = options.apiKey;
-      }
+    // Create multipart form data boundary
+    const boundary =
+      "----LangSmithFormBoundary" + Math.random().toString(36).slice(2);
+
+    const isNodeFetch = _globalFetchImplementationIsNodeFetch();
+    const buildBuffered = () => this._createNodeFetchBody(parts, boundary);
+    const buildStream = () => this._createMultipartStream(parts, boundary);
 
-      const res = await this.batchIngestCaller.call(
+    const send = async (body: BodyInit) =>
+      await this.batchIngestCaller.call(
         _getFetchImplementation(this.debug),
         `${options?.apiUrl ?? this.apiUrl}/runs/multipart`,
         {
           method: "POST",
-          headers,
+          headers: {
+            ...this.headers,
+            "Content-Type": `multipart/form-data; boundary=${boundary}`,
+          },
           body,
           duplex: "half",
           signal: AbortSignal.timeout(this.timeout_ms),
           ...this.fetchOptions,
         }
       );
+
+    try {
+      let streamedAttempt = false;
```

Comment:
I think we should set this for the client so that it never tries again, since I don't really think this is retryable